include/rocksdb/merge_operator.h

```diff
@@ -190,6 +190,15 @@ class MergeOperator {
   // PartialMerge/PartialMergeMulti should be implemented accordingly to handle
   // a single operand.
   virtual bool AllowSingleOperand() const { return false; }
+
+  // Allows to control when to invoke a full merge during Get.
+  // This could be used to limit the number of merge operands that are looked at
+  // during a point lookup, thereby helping in limiting the number of levels to
+  // read from.
+  // Doesn't help with iterators.
+  virtual bool ShouldMerge(const std::vector<Slice>& operands) const {
```

Comment:
one question, when this gets called, do we already read all the operands from the disk? which means we still need to pay the disk io?

Reply:
No. This `ShouldMerge` callback function gets called for each merge operand as we go down the levels, and we don't read more operands when `ShouldMerge` returns true. So we don't pay for any more disk io than what is absolutely necessary. 

Example: Lets say:
```
bool ShouldMerge(const std::vector<Slice>& operands) const {
  return (operands.size() >= 2);
}
```
Lets say, a key k1 has 8 merge operands spread over different levels.
On encountering a Get(k1), `ShouldMerge` gets called with one operand first, and returns false. 
Next it gets called when the second operand is encountered, returns true, which leads to invoking the FullMerge operator, and a value is returned. 
All the merge operands for k1 at higher levels are not read from disk.